src/odemis/gui/cont/multi_point_correlation.py

```diff
@@ -0,0 +1,762 @@
+
+    @call_in_wx_main
+    def _on_current_coordinates_changes(self, coordinates) -> None:
+        """Upaate the coordinates of the current target in the grid and update the correlation result."""
+        target = self._tab_data_model.main.currentTarget.value
+        self.current_target_coordinate_subscription = False
+        temp_check = False
+        for row in range(self.grid.GetNumberRows()):
+            if self._selected_target_in_grid(target, row):
+                if target.name.value.startswith("FIB"):
+                    pixel_coords = self.correlation_target.fib_stream.getPixelCoordinates(
+                        (target.coordinates.value[0], target.coordinates.value[1]), check_bbox=False)
+                else:
+                    pixel_coords = self.correlation_target.fm_streams[0].getPixelCoordinates(
+                        (target.coordinates.value[0], target.coordinates.value[1]), check_bbox=False)
+                    if (self.grid.GetCellValue(row,
+                                               GridColumns.Z.value)) != f"{float(target.coordinates.value[2]):.{GRID_PRECISION}f}":
+                        temp_check = True
+                    self.grid.SetCellValue(row, GridColumns.Z.value,
+                                           f"{target.coordinates.value[2]:.{GRID_PRECISION}f}")
+                # Get cell value
+                if (self.grid.GetCellValue(row, GridColumns.X.value) != f"{pixel_coords[0]:.{GRID_PRECISION}f}" or
+                        self.grid.GetCellValue(row, GridColumns.Y.value) != f"{pixel_coords[1]:.{GRID_PRECISION}f}"):
+                    temp_check = True
+                self.grid.SetCellValue(row, GridColumns.X.value, f"{pixel_coords[0]:.{GRID_PRECISION}f}")
+                self.grid.SetCellValue(row, GridColumns.Y.value, f"{pixel_coords[1]:.{GRID_PRECISION}f}")
+                self._update_feature_correlation_target()
+
+            if self.check_correlation_conditions() and temp_check:
+                self._queue_latest_change()
+
+    @call_in_wx_main
+    def _on_target_changes(self, targets) -> None:
+        if not self.correlation_target.fm_streams and self.previous_group:
+            for key, indices in self.stream_groups.items():
+                if key == self.previous_group:
+                    self.correlation_target.fm_stream_key = key
+                    for index in indices:
+                        stream = self.streams_list[index]
+                        self.correlation_target.fm_streams.append(stream)
+                else:
+                    for index in indices:
+                        stream = self.streams_list[index]
+                        # FIB static stream should not be removed
+                        if isinstance(stream, StaticFluoStream):
+                             self._panel.streambar_controller.removeStreamPanel(stream)
+
+        # Clear the grid before populating it with new data
+        self.grid.ClearGrid()
```

Comment:
Based on my experience this will be computationally expensive in the future for very large number of targets.

I know you grid usage is different to mine, having said that I store the grid data https://github.com/delmic/odemis/blob/d877433424fc4e93382dbb67951e6c0b841cd65a/src/odemis/gui/comp/fastem_project_manager_panel.py#L568 per project